### PR TITLE
Add more ClientConfig Options for Object Store RequestBuilder (#3127)

### LINKED ISSUE #3127: Object Store: Allow custom reqwest client
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

I am connecting to our Ceph cluster and getting intermittent connection closed errors from the server. I have tried both multipart and non-multipart uploads and have had the same issues.

Others have run into this issue with the S3 API with reqwest/hyper and have needed to change their client configuration to resolve the issue.
See:
* https://github.com/hyperium/hyper/issues/2136
* https://github.com/hyperium/hyper/issues/2136#issuecomment-861826148

**Describe the solution you'd like**

I would like to expose the reqwest client for custom configuration.

**Describe alternatives you've considered**

An alternative would be to expose individual configuration settings in the Reqwest client and continue to shield creation of the client from the end user. This seems like unnecessary complexity for the object store to expose.

Another alternative would be to make the reqwest client public so that it can be modified after the S3Client has been created. Exposing the client in this way is bad API design. Keeping the client private while allowing initial configuration seems to me to be the best route.

**Additional context**

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #3127

# Rationale for this change
 

Adds more configuration options for the HTTP client. In future we may want to tweak some of the defaults to work better with common object stores, in particular the keep alive timeouts may be higher than they should be without keep alive pings. With this PR we can find a config that works, and then bake that

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: object_store/src/client/mod.rs [hunk 1]
```diff
@@ -33,11 +35,25 @@ fn map_client_error(e: reqwest::Error) -> super::Error {
     }
 }
 
+static DEFAULT_USER_AGENT: &str =
```
Comment (tustvold) [inline object_store/src/client/mod.rs:38]: Changing the user-agent is not a breaking change, especially given that reqwest's default include its patch version

**Review by alamb — APPROVED** [review]
LGTM -- thanks @tustvold

**Comment by declark1** [comment]
What about an option for http2_max_frame_size?

**Comment by tustvold** [comment]
I'd be happy to review a PR that added this

**Pull request merged**